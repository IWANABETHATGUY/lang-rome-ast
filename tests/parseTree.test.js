import { describe, it, expect } from "vitest";
import { getLezerSyntaxTree } from "lezer-syntax-tree-debug";
import { parser } from "../dist/index";
import * as fs from "fs";
import * as path from "path";

describe("basic", () => {
	fs.readdirSync(path.resolve(__dirname, "./fixtures")).forEach((file) => {
		it(file, () => {
			let source = fs.readFileSync(
				path.resolve(__dirname, "./fixtures", file),
				"utf8",
			);
			let syntaxTree = parser.parse(source);

			let syntaxTreeDebug = getLezerSyntaxTree(syntaxTree.topNode, source);
			expect(syntaxTreeDebug).toMatchSnapshot();
		});
	});
});
