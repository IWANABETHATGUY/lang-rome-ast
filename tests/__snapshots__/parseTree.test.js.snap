// Vitest Snapshot v1

exports[`basic > optional_syntax_node_type_of_syntax_node_list.txt 1`] = `
"Program @0..387 ''
  SyntaxNode @0..387 ''
    SyntaxNodeType @0..18 'JsUnknownParameter'
    SyntaxMemberList @25..385 ''
      SyntaxNodePair @25..384 ''
        SyntaxNodePairKey @25..30 'items'
        SyntaxNodeList @32..384 ''
          SyntaxNode @42..377 ''
            SyntaxNodeType @42..58 'TsTypeAnnotation'
            SyntaxMemberList @73..367 ''
              SyntaxNodePair @73..125 'colon_token: COLON@119..121 ":" [] [Whitespace(" ")]'
                SyntaxNodePairKey @73..84 'colon_token'
                SyntaxToken @86..125 'COLON@119..121 ":" [] [Whitespace(" ")]'
                  SyntaxTokenType @86..91 'COLON'
                  Range @92..100 '119..121'
                    Number @92..95 '119'
                    Number @97..100 '121'
                  String @101..104 '":"'
                  TriviaPart @105..107 '[]'
                  TriviaPart @108..125 '[Whitespace(" ")]'
                    TriviaList @109..124 'Whitespace(" ")'
                      Trivia @109..124 'Whitespace(" ")'
                        Identifier @109..119 'Whitespace'
                        String @120..123 '" "'
              SyntaxNodePair @139..366 ''
                SyntaxNodePairKey @139..141 'ty'
                SyntaxNode @143..366 ''
                  SyntaxNodeType @143..158 'TsReferenceType'
                  SyntaxMemberList @177..352 ''
                    SyntaxNodePair @177..299 ''
                      SyntaxNodePairKey @177..181 'name'
                      SyntaxNode @183..299 ''
                        SyntaxNodeType @183..204 'JsReferenceIdentifier'
                        SyntaxMemberList @227..281 'value_token: IDENT@121..138 "ErrorsTextOptions" [] [],'
                          SyntaxNodePair @227..280 'value_token: IDENT@121..138 "ErrorsTextOptions" [] []'
                            SyntaxNodePairKey @227..238 'value_token'
                            SyntaxToken @240..280 'IDENT@121..138 "ErrorsTextOptions" [] []'
                              SyntaxTokenType @240..245 'IDENT'
                              Range @246..254 '121..138'
                                Number @246..249 '121'
                                Number @251..254 '138'
                              String @255..274 '"ErrorsTextOptions"'
                              TriviaPart @275..277 '[]'
                              TriviaPart @278..280 '[]'
                    SyntaxNodePair @317..351 'type_arguments: missing (optional)'
                      SyntaxNodePairKey @317..331 'type_arguments'
                      MissingOptional @333..351 'missing (optional)'"
`;
